File: 0200_Number of Islands/NumberofIslands.java
class Solution {
    public int numIslands(char[][] grid) {
        int numIslands = 0;
        if (grid == null)
            return numIslands;
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[i].length; j++) {
                if (grid[i][j] == '1') {
                    numIslands++;
                    check(grid, i, j, grid.length, grid[i].length);
                }
            }
        }

        return numIslands;
    }

    private void check(char[][] grid, int x, int y, int rowLen, int colLen) {
        if (x < 0 || y < 0 || x >= rowLen || y >= colLen)
            return;
        if (grid[x][y] == '1') {
            grid[x][y] = '0';
            check(grid, x - 1, y, rowLen, colLen);
            check(grid, x, y - 1, rowLen, colLen);
            check(grid, x + 1, y, rowLen, colLen);
            check(grid, x, y + 1, rowLen, colLen);

        }
    }
}

File: 0200_Number of Islands/NumberofIslands.cs
public class Solution {
    public int NumIslands(char[][] grid) {
        if(grid == null) return 0;
        var ans = 0;
        for(int i=0;i<grid.Length;i++){
            for(int j=0;j<grid[i].Length;j++){
                if(grid[i][j] == '1'){
                    ans++;
                    Scan(grid,i,j);
                }
            }
        }
        
        return ans;
    }
    
    private void Scan(char[][] grid, int x, int y){
        if(x < 0 || y < 0 || x >= grid.Length || y >=  grid[0].Length){
            return ; 
        }
        
        if(grid[x][y] == '0') return ;
        grid[x][y] = '0';
        Scan(grid, x-1, y);
        Scan(grid, x+1, y);
        Scan(grid, x, y-1);
        Scan(grid, x, y+1);
    }
}